### PR TITLE
Use secrets for credentials for Hetzner

### PR DESCRIPTION
**Pull request by nikhita**
Follow up of https://github.com/kubermatic/kubermatic/pull/3836
/hold

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nikhita** [comment]
This is ready for review.

**Comment by nikhita** [comment]
Ready for another round of review.

File: api/pkg/resources/resources.go [hunk 1]
```diff
@@ -364,6 +364,8 @@ const (
 	AWSAccessKeyID     = "accessKeyId"
 	AWSSecretAccessKey = "secretAccessKey"
 
+	HetznerToken = "hetzner"
```
Comment (alvaroaleman) [inline api/pkg/resources/resources.go]: Can we change the value of this to `token`?

Comment (nikhita) [reply api/pkg/resources/resources.go]: Yikes, fixed.

**Comment by nikhita** [comment]
/assign @zreigz 
for approval

**Comment by nikhita** [comment]
All tests are green :)

**Comment by nikhita** [comment]
/hold
until the test is fixed

**Comment by alvaroaleman** [comment]
_Supposively_ everything will work now.
/hold cancel
/retest

**Pull request merged**